ui/component/or-app/src/index.ts

```diff
@@ -283,15 +298,27 @@ export class OrApp<S extends AppStateKeyed> extends LitElement {
             return;
         }
 
-        if (changedProps.has("_page")) {
+        // If either page or 'offline'-status is changed, it should update to the correct page,
+        // by appending the page to the HTML content
+        if (changedProps.has("_page") || changedProps.has("_offline")) {
             if (this._mainElem) {
                 if (this._mainElem.firstElementChild) {
                     this._mainElem.firstElementChild.remove();
                 }
                 if (this._page) {
                     const pageProvider = this.appConfig!.pages.find((page) => page.name === this._page);
                     if (pageProvider) {
-                        const pageElem = pageProvider.pageCreator();
+                        let pageElem;
+                        if(!this._offline || pageProvider.allowOffline) {
+                            pageElem = pageProvider.pageCreator();
+                        } else {
+                            // Check for override offline page
+                            if(this.appConfig?.offlinePage) {
+                                pageElem = this.appConfig.offlinePage;
```

Comment:
Does custom offlinePage not need access to the `store`?

Reply:
No, it only listens to manager events (`OrEvent`).
But might be in the future, so I changed `page-offline` to a `PageProvider`.